src/picosvg/svg.py

```diff
@@ -359,14 +440,27 @@ def resolve_use(self, inplace=False):
         self._resolve_use(self.svg_root)
         return self
 
-    def _resolve_clip_path(self, clip_path_url) -> SVGPath:
+    def _resolve_clip_path(
+        self, clip_path_url, transform=Affine2D.identity()
+    ) -> SVGPath:
         clip_path_el = self.resolve_url(clip_path_url, "clipPath")
         self._resolve_use(clip_path_el)
-        self._ungroup(clip_path_el)
 
-        # union all the shapes under the clipPath
-        # Fails if there are any non-shapes under clipPath
-        return SVGPath.from_commands(union([from_element(e) for e in clip_path_el]))
+        transform = _element_transform(clip_path_el, transform)
```

Comment:
hm I'm not sure you want to transform the clipPath's `transform` and then its individual children's `transform`s indipendently from one another. I believe the correct thing to do would be, if the clipPath container itself defines a `transform` push that down to the children first (concatenating with the children's transform), and each children's transform with the current transform

Reply:
I'm not quite sure I follow. We pull the clipPath transform above, merge it with any child transform, and then transform the child (below). 

_element_transform is specifically meant for that pattern; you march down the graph and each time you encounter a transform you update the active transform.

Did I muddle that up?

Reply:
actually you're right, I somehow missed the fact that you were re-assigning the local variable `transform`.